pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/parquet/ParquetFileProcessor.java

```diff
@@ -224,14 +230,25 @@ void processFile(FileNameWithOffset fileNameWithBeginOffset, long firstSequenceN
     void deleteCompletedFiles() throws Exception {
         final List<FileNameWithOffset> completedFiles = state.getCompletedFiles();
         completedFiles.forEach(file -> {
-            try {
-                Files.deleteIfExists(Paths.get(file.fileName));
-                log.info("deleteCompletedFiles: Deleted file {}", file.fileName);
-                // Only remove from database if we could delete file.
-                state.deleteCompletedFile(file.fileName);
+             //Obtain a lock on file
+            try(FileChannel channel = FileChannel.open(Paths.get(file.fileName),StandardOpenOption.WRITE)){
+                try(FileLock lock = channel.tryLock()) {
+                    if(lock!=null){
+                        Files.deleteIfExists(Paths.get(file.fileName));
+                        log.info("deleteCompletedFiles: Deleted file {}", file.fileName);
+                        lock.release();
+                        // Only remove from database if we could delete file.
+                        state.deleteCompletedFile(file.fileName);                        
+                    }
+                    else{
+                        log.info("Unable to obtain lock");
+                        throw new Exception();
+                    }
+                }
             } catch (Exception e) {
-                log.warn("Unable to delete ingested file {}", e);
-                // We can continue on this error. It will be retried on the next iteration.
+                log.warn("Unable to delete ingested file {}", e.getMessage());
+                log.warn("File is locked by another process. Will retry deletion.");
```

Comment:
This log line will be written for any exception not just when file is locked. 
This is not correct we should have this message only when it is really locked by other process.

Reply:
I've modified it.